services/api-server/src/simcore_service_api_server/api/routes/studies.py

```diff
@@ -0,0 +1,39 @@
+""" Studies collections
+
+
+"""
+import logging
+from typing import Any
+
+from fastapi import APIRouter, Depends
+
+from ...core.settings import BasicSettings
+from ...models.schemas.studies import StudyID, StudyPort
+from ..dependencies.webserver import AuthSession, get_webserver_session
+
+logger = logging.getLogger(__name__)
+router = APIRouter()
+settings = BasicSettings.create_from_envs()
+
+#
+# studies/{project_id}
+#
+
+
+@router.get(
+    "/{study_id}/ports",
+    response_model=list[StudyPort],
+    include_in_schema=settings.API_SERVER_DEV_FEATURES_ENABLED,
+)
+async def list_study_ports(
+    study_id: StudyID,
+    webserver_api: AuthSession = Depends(get_webserver_session),
+):
+    """Lists metadata on ports of a given study
+
+    New in *version 0.5.0* (only with API_SERVER_DEV_FEATURES_ENABLED=1)
```

Comment:
Q: do you have a way to define in which version a route is available? that could be cool to have a decorator ```@version0.5.0```

Reply:
The idea here is to document when this function was available in the api (similar to the python doc). For the moment I do it manually but a decorator would definitively be a better way to handle this. Will create an issue for this

Reply:
Might I suggest at this point to have it as in the python docs. Worded like: `added in version 0.5.0`?